### ISSUE TITLE
[Reflection] Crash on startup relating to style/language

### ISSUE THREAD

**Issue by charlesroddie** [xamarin/xamarin-android#4624]
We are getting an error on startup when the system language is set to Chinese instead of English. The error message reports a reflection error so the ultimate cause is lack of safety resulting from use of reflection.

```
System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation.
System.TypeInitializationException: The type initializer for 'Style' threw an exception.
```
Full error message:
```
Xamarin Exception Stack:
System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation.
  at System.Reflection.RuntimeConstructorInfo.InternalInvoke (System.Object obj, System.Object[] parameters, System.Boolean wrapExceptions) [0x0001d] in <5429ce7d570b46cb9c509d14b217d379>:0
  at System.Reflection.RuntimeConstructorInfo.DoInvoke (System.Object obj, System.Reflection.BindingFlags invokeAttr, System.Reflection.Binder binder, System.Object[] parameters, System.Globalization.CultureInfo culture) [0x00086] in <5429ce7d570b46cb9c509d14b217d379>:0
  at System.Reflection.RuntimeConstructorInfo.Invoke (System.Reflection.BindingFlags invokeAttr, System.Reflection.Binder binder, System.Object[] parameters, System.Globalization.CultureInfo culture) [0x00000] in <5429ce7d570b46cb9c509d14b217d379>:0
  at System.RuntimeType.CreateInstanceImpl (System.Reflection.BindingFlags bindingAttr, System.Reflection.Binder binder, System.Object[] args, System.Globalization.CultureInfo culture, System.Object[] activationAttributes, System.Threading.StackCrawlMark& stackMark) [0x0022b] in <5429ce7d570b46cb9c509d14b217d379>:0
  at System.Activator.CreateInstance (System.Type type, System.Reflection.BindingFlags bindingAttr, System.Reflection.Binder binder, System.Object[] args, System.Globalization.CultureInfo culture, System.Object[] activationAttributes) [0x0009c] in <5429ce7d570b46cb9c509d14b217d379>:0
  at System.Activator.CreateInstance (System.Type type, System.Object[] args) [0x00000] in <5429ce7d570b46cb9c509d14b217d379>:0
  at Xamarin.Forms.Internals.DependencyResolver.ResolveOrCreate (System.Type type, System.Object source, System.Type visualType, System.Object[] args) [0x000af] in <46bb4f43387e4d3aa4c7b3d5909656ed>:0
  at Xamarin.Forms.Internals.Registrar`1[TRegistrable].GetHandler (System.Type type, System.Object source, Xamarin.Forms.IVisual visual, System.Object[] args) [0x00060] in <46bb4f43387e4d3aa4c7b3d5909656ed>:0
  at Xamarin.Forms.Internals.Registrar`1[TRegistrable].GetHandlerForObject[TOut] (System.Object obj, System.Object[] args) [0x00041] in <46bb4f43387e4d3aa4c7b3d5909656ed>:0
  at Xamarin.Forms.Platform.Android.Platform.CreateRenderer (Xamarin.Forms.VisualElement element, Android.Content.Context context) [0x00005] in <78df70a8772c4bef8b9762ec32567ea7>:0
  at Xamarin.Forms.Platform.Android.VisualElementPackager.AddChild (Xamarin.Forms.VisualElement view, Xamarin.Forms.Platform.Android.IVisualElementRenderer oldRenderer, Xamarin.Forms.Platform.Android.RendererPool pool, System.Boolean sameChildren) [0x000c0] in <78df70a8772c4bef8b9762ec32567ea7>:0
  at Xamarin.Forms.Platform.Android.VisualElementPackager.SetElement (Xamarin.Forms.VisualElement oldElement, Xamarin.Forms.VisualElement newElement) [0x0016c] in <78df70a8772c4bef8b9762ec32567ea7>:0
  at Xamarin.Forms.Platform.Android.VisualElementPackager.Load () [0x00000] in <78df70a8772c4bef8b9762ec32567ea7>:0
  at Xamarin.Forms.Platform.Android.VisualElementRenderer`1[TElement].SetPackager (Xamarin.Forms.Platform.Android.VisualElementPackager packager) [0x00007] in <78df70a8772c4bef8b9762ec32567ea7>:0
  at Xamarin.Forms.Platform.Android.VisualElementRenderer`1[TElement].SetElement (TElement element) [0x000dd] in <78df70a8772c4bef8b9762ec32567ea7>:0
  at Xamarin.Forms.Platform.Android.VisualElementRenderer`1[TElement].Xamarin.Forms.Platform.Android.IVisualElementRenderer.SetElement (Xamarin.Forms.VisualElement element) [0x00033] in <78df70a8772c4bef8b9762ec32567ea7>:0
  at Xamarin.Forms.Platform.Android.Platform.CreateRenderer (Xamarin.Forms.VisualElement element, Android.Content.Context context) [0x0001f] in <78df70a8772c4bef8b9762ec32567ea7>:0
  at Xamarin.Forms.Platform.Android.VisualElementPackager.AddChild (Xamarin.Forms.VisualElement view, Xamarin.Forms.Platform.Android.IVisualElementRenderer oldRenderer, Xamarin.Forms.Platform.Android.RendererPool pool, System.Boolean sameChildren) [0x000c0] in <78df70a8772c4bef8b9762ec32567ea7>:0
  at Xamarin.Forms.Platform.Android.VisualElementPackager.SetElement (Xamarin.Forms.VisualElement oldElement, Xamarin.Forms.VisualElement newElement) [0x0016c] in <78df70a8772c4bef8b9762ec32567ea7>:0
  at Xamarin.Forms.Platform.Android.VisualElementPackager.Load () [0x00000] in <78df70a8772c4bef8b9762ec32567ea7>:0
  at Xamarin.Forms.Platform.Android.VisualElementRenderer`1[TElement].SetPackager (Xamarin.Forms.Platform.Android.VisualElementPackager packager) [0x00007] in <78df70a8772c4bef8b9762ec32567ea7>:0
  at Xamarin.Forms.Platform.Android.VisualElementRenderer`1[TElement].SetElement (TElement element) [0x000dd] in <78df70a8772c4bef8b9762ec32567ea7>:0
  at Xamarin.Forms.Platform.Android.VisualElementRenderer`1[TElement].Xamarin.Forms.Platform.Android.IVisualElementRenderer.SetElement (Xamarin.Forms.VisualElement element) [0x00033] in <78df70a8772c4bef8b9762ec32567ea7>:0
  at Xamarin.Forms.Platform.Android.Platform.CreateRenderer (Xamarin.Forms.VisualElement element, Android.Content.Context context) [0x0001f] in <78df70a8772c4bef8b9762ec32567ea7>:0
  at Xamarin.Forms.Platform.Android.VisualElementPackager.AddChild (Xamarin.Forms.VisualElement view, Xamarin.Forms.Platform.Android.IVisualElementRenderer oldRenderer, Xamarin.Forms.Platform.Android.RendererPool pool, System.Boolean sameChildren) [0x000c0] in <78df70a8772c4bef8b9762ec32567ea7>:0
  at Xamarin.Forms.Platform.Android.VisualElementPackager.SetElement (Xamarin.Forms.VisualElement oldElement, Xamarin.Forms.VisualElement newElement) [0x0016c] in <78df70a8772c4bef8b9762ec32567ea7>:0
  at Xamarin.Forms.Platform.Android.VisualElementPackager.Load () [0x00000] in <78df70a8772c4bef8b9762ec32567ea7>:0
  at Xamarin.Forms.Platform.Android.VisualElementRenderer`1[TElement].SetPackager (Xamarin.Forms.Platform.Android.VisualElementPackager packager) [0x00007] in <78df70a8772c4bef8b9762ec32567ea7>:0
  at Xamarin.Forms.Platform.Android.VisualElementRenderer`1[TElement].SetElement (TElement element) [0x000dd] in <78df70a8772c4bef8b9762ec32567ea7>:0
  at Xamarin.Forms.Platform.Android.VisualElementRenderer`1[TElement].Xamarin.Forms.Platform.Android.IVisualElementRenderer.SetElement (Xamarin.Forms.VisualElement element) [0x00033] in <78df70a8772c4bef8b9762ec32567ea7>:0
  at Xamarin.Forms.Platform.Android.Platform.CreateRenderer (Xamarin.Forms.VisualElement element, Android.Content.Context context) [0x0001f] in <78df70a8772c4bef8b9762ec32567ea7>:0
  at Xamarin.Forms.Platform.Android.AppCompat.Platform.AddChild (Xamarin.Forms.Page page, System.Boolean layout) [0x0000d] in <78df70a8772c4bef8b9762ec32567ea7>:0
  at Xamarin.Forms.Platform.Android.AppCompat.Platform.SetPageInternal (Xamarin.Forms.Page newRoot) [0x00061] in <78df70a8772c4bef8b9762ec32567ea7>:0
  at Xamarin.Forms.Platform.Android.AppCompat.Platform.SetPage (Xamarin.Forms.Page newRoot) [0x000e6] in <78df70a8772c4bef8b9762ec32567ea7>:0
  at Xamarin.Forms.Platform.Android.FormsAppCompatActivity.InternalSetPage (Xamarin.Forms.Page page) [0x0003f] in <78df70a8772c4bef8b9762ec32567ea7>:0
  at Xamarin.Forms.Platform.Android.FormsAppCompatActivity.SetMainPage () [0x0000c] in <78df70a8772c4bef8b9762ec32567ea7>:0
  at Xamarin.Forms.Platform.Android.FormsAppCompatActivity.LoadApplication (Xamarin.Forms.Application application) [0x00140] in <78df70a8772c4bef8b9762ec32567ea7>:0
  at MathSpire.Droid.MainActivity.OnCreate (Android.OS.Bundle bundle) [0x00056] in <3caeb3a1b07b4502b21596e5125ecfc8>:0
  at Android.App.Activity.n_OnCreate_Landroid_os_Bundle_ (System.IntPtr jnienv, System.IntPtr native__this, System.IntPtr native_savedInstanceState) [0x00011] in <0c0eefeeda0e45a8ba05827ca2cae106>:0
  at (wrapper dynamic-method) Android.Runtime.DynamicMethodNameCounter.15(intptr,intptr,intptr)
--- End of inner exception stack trace ---
  System.TypeInitializationException: The type initializer for 'Style' threw an exception.
    at (wrapper managed-to-native) System.Object.__icall_wrapper_mono_generic_class_init(intptr)
    at Xamarin.Forms.Platform.Android.ScrollViewRenderer..ctor (Android.Content.Context context) [0x00000] in <78df70a8772c4bef8b9762ec32567ea7>:0
    at (wrapper managed-to-native) System.Reflection.RuntimeConstructorInfo.InternalInvoke(System.Reflection.RuntimeConstructorInfo,object,object[],System.Exception&)
    at System.Reflection.RuntimeConstructorInfo.InternalInvoke (System.Object obj, System.Object[] parameters, System.Boolean wrapExceptions) [0x00005] in <5429ce7d570b46cb9c509d14b217d379>:0
  --- End of inner exception stack trace ---
    System.IO.FileNotFoundException: Could not load file or assembly 'netstandard, Version=2.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51' or one of its dependencies.
      at (wrapper managed-to-native) System.MonoCustomAttrs.GetCustomAttributesInternal(System.Reflection.ICustomAttributeProvider,System.Type,bool)
      at System.MonoCustomAttrs.GetCustomAttributesBase (System.Reflection.ICustomAttributeProvider obj, System.Type attributeType, System.Boolean inheritedOnly) [0x00013] in <5429ce7d570b46cb9c509d14b217d379>:0
      at System.MonoCustomAttrs.GetCustomAttributes (System.Reflection.ICustomAttributeProvider obj, System.Type attributeType, System.Boolean inherit) [0x00037] in <5429ce7d570b46cb9c509d14b217d379>:0
      at System.Reflection.RuntimeAssembly.GetCustomAttributes (System.Type attributeType, System.Boolean inherit) [0x00000] in <5429ce7d570b46cb9c509d14b217d379>:0
      at Android.Runtime.ResourceIdManager.GetResourceTypeFromAssembly (System.Reflection.Assembly assembly) [0x00000] in <0c0eefeeda0e45a8ba05827ca2cae106>:0
      at Android.Runtime.ResourceIdManager.UpdateIdValues () [0x00044] in <0c0eefeeda0e45a8ba05827ca2cae106>:0
      at Xamarin.Forms.Platform.Android.Resource+Style..cctor () [0x0139c] in <78df70a8772c4bef8b9762ec32567ea7>:0
```

**Comment by radekdoulik** [comment]
Did you added CJK encoding in the Android Options properties page? If you did could you please attach a small project which would replicate the issue?

**Comment by charlesroddie** [comment]
Thanks for the advice. I've added CJK and will see if that helps and report back. (The app will not need to display non-Western characters. Just needs to work in English on non-English devices.)

**Comment by charlesroddie** [comment]
@foggyfinder was able to reproduce this and says it seems to be fixed in XF 4.6.

**Comment by radekdoulik** [comment]
Thanks for the information.
